extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/EventBusInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,160 @@
+        return producerInstrumenter;
+    }
+
+    static Instrumenter<Message, Message> getConsumerInstrumenter(final OpenTelemetry openTelemetry) {
+        EventBusAttributesExtractor eventBusAttributesExtractor = new EventBusAttributesExtractor(RECEIVE);
+
+        InstrumenterBuilder<Message, Message> serverBuilder = Instrumenter.builder(
+                openTelemetry,
+                INSTRUMENTATION_NAME, MessagingSpanNameExtractor.create(eventBusAttributesExtractor));
+
+        return serverBuilder
+                .addAttributesExtractor(eventBusAttributesExtractor)
+                .newConsumerInstrumenter(new TextMapGetter<>() {
+                    @Override
+                    public Iterable<String> keys(final Message message) {
+                        return message.headers().names();
+                    }
+
+                    @Override
+                    public String get(final Message message, final String key) {
+                        if (message == null) {
+                            return null;
+                        }
+                        return message.headers().get(key);
+                    }
+                });
+    }
+
+    static Instrumenter<Message, Message> getProducerInstrumenter(final OpenTelemetry openTelemetry) {
+        EventBusAttributesExtractor eventBusAttributesExtractor = new EventBusAttributesExtractor(SEND);
+
+        InstrumenterBuilder<Message, Message> serverBuilder = Instrumenter.builder(
+                openTelemetry,
+                INSTRUMENTATION_NAME, MessagingSpanNameExtractor.create(eventBusAttributesExtractor));
+
+        return serverBuilder
+                .addAttributesExtractor(eventBusAttributesExtractor)
+                .newProducerInstrumenter((message, key, value) -> {
+                    if (message != null) {
+                        message.headers().set(key, value);
+                    }
+                });
+    }
+
+    private static class EventBusAttributesExtractor extends MessagingAttributesExtractor<Message, Message> {
+        private final MessageOperation operation;
+
+        public EventBusAttributesExtractor(final MessageOperation operation) {
+            this.operation = operation;
+        }
+
+        @Override
+        public MessageOperation operation() {
+            return operation;
+        }
+
+        @Override
+        protected String system(final Message message) {
+            return "vert.x";
+        }
+
+        @Override
+        protected String destinationKind(final Message message) {
+            return "topic";
+        }
+
+        @Override
+        protected String destination(final Message message) {
+            return message.address();
+        }
+
+        @Override
+        protected boolean temporaryDestination(final Message message) {
+            return false;
+        }
+
+        @Override
+        protected String protocol(final Message message) {
+            return null;
```

Comment:
It could be any known protocol like AMQP, MQTT, etc. In this case, since this is handmade, it is acceptable to leave this without a value (it is not required per the OTel spec).

It is correct to name this the `Vert.x` protocol? :)

Reply:
yes, it's vert.x specific, and while being very simple, I'm not aware of any document describing it (except the code)